tests/test_causal_model.py

```diff
@@ -144,6 +142,59 @@ def test_graph_input2(self, beta, num_instruments, num_samples, num_treatments):
         common_causes = model.get_common_causes()
         assert all(node_name in common_causes for node_name in ["X1", "X2"])
 
+    @pytest.mark.parametrize(["beta", "num_instruments", "num_samples", "num_treatments"],
+                             [(10, 1, 100, 1),])
+    def test_graph_input3(self, beta, num_instruments, num_samples, num_treatments):
+        num_common_causes = 5
+        data = dowhy.datasets.linear_dataset(beta=beta,
+                                             num_common_causes=num_common_causes,
+                                             num_instruments=num_instruments,
+                                             num_samples=num_samples,
+                                             num_treatments = num_treatments,
+                                             treatment_is_binary=True)
+
+        model = CausalModel(
+            data=data['df'],
+            treatment=data["treatment_name"],
+            outcome=data["outcome_name"],
+            graph=data["gml_graph"],
+            proceed_when_unidentifiable=True,
+            test_significance=None
+        )
+        # removing two common causes 
+        gml_str = """dag {
+        "Unobserved Confounders" [pos="0.491,-1.056"]
+        X0 [pos="-2.109,0.057"]
+        X1 [pos="-0.453,-1.562"]
+        X2 [pos="-2.268,-1.210"]
+        Z0 [pos="-1.918,-1.735"]
+        v0 [pos="-1.525,-1.293"]
+        y [pos="-1.164,-0.116"]
+        "Unobserved Confounders" -> v0
+        "Unobserved Confounders" -> y
+        X0 -> v0
+        X0 -> y
+        X1 -> v0
+        X1 -> y
+        X2 -> v0
+        X2 -> y
+        Z0 -> v0
+        v0 -> y
+        }
+        """
+        print(gml_str)
+        model = CausalModel(
+            data=data['df'],
+            treatment=data["treatment_name"],
+            outcome=data["outcome_name"],
+            graph=gml_str,
+            proceed_when_unidentifiable=True,
+            test_significance=None,
+            missing_nodes_as_confounders=True
+        )
+        common_causes = model.get_common_causes()
+        assert all(node_name in common_causes for node_name in ["X1", "X2"])
```

Comment:
can add more assertions here. you can use `model._graph.get_all_nodes` and make sure that the all the nodes are there.

Reply:
The new updates are good. I will suggest one more test: obtain `model._graph.get_all_nodes(include_unobserved=False)` and check that it does not include "unobserved confounders".